app/core/abilities/abilities.ts

```diff
@@ -1934,31 +1934,22 @@ export class AccelerockStrategy extends AbilityStrategy {
     target: PokemonEntity,
     crit: boolean
   ) {
-    super.process(pokemon, state, board, target, crit)
-    const farthestCoordinate =
+    const destination =
       board.getFarthestTargetCoordinateAvailablePlace(pokemon)
-
-    if (farthestCoordinate) {
-      pokemon.moveTo(farthestCoordinate.x, farthestCoordinate.y, board)
-      const targetCoordinates = state.getNearestTargetAtRangeCoordinates(
+    target = destination?.target ?? target
```

Comment:
if no destination is found, the fallback to target wont be useful because of the if(destination) condition

maybe the target.handleSpecialDamage() should be outside of that condition ?